### DISCUSSION TITLE
Can I mix Talos with other distros in a K8S cluster?

### DISCUSSION THREAD

**Discussion by abceleung** [siderolabs/talos#9435] [Q&A] [answered]
(Newbie question, never setup a K8S cluster before)

I have two x86 machines and a Raspberry Pi 5. I was planning to install Talos on all of them, but I just found out that Raspberry Pi 5 is not supported by almost all distros, including Talos.

I think I have to install Raspbian on the Pi and setup K8S manually. Would the K8S setup still work if I install Talos on the other two x86 machines?

**Comment by smira** [comment, marked as answer]
Talos has a requirement that Talos machines can only join Talos controlplane, but workers can be set up e.g. using `kubeadm`. Some features of Talos might not work with non-Talos machines in the mix (e.g. KubeSpan), but the Kubernetes itself should be fine.

See [this issue](https://github.com/siderolabs/talos/issues/3990) for details.
